### ISSUE TITLE
Auto-focus the searchbar?

### ISSUE THREAD

**Issue by sinkingshriek** [npm/newww#1039]
I'm being really picky here and I'm aware that there's a way to do this which is mentioned [here](https://github.com/npm/newww/issues/513) but for those who don't know this trick it'd be really helpful if the search was auto-focused just like how google does it because the number 1 reason for me to come to npmjs.com is to search for packages. 
It's also a really simple change which just needs `autofocus` attribute [here](https://github.com/npm/newww/blob/master/templates/layouts/default.hbs#L96) so sending a PR just for one word sounded a little ridiculous to me. Let me know how you guys feel about this. 
Again, I'm being extremely picky but it's nice to have for new comers.

**Comment by isaacs** [comment]
Roughly 50% of users believe that the search bar should be auto-focused, and don't see how anyone could ever want it otherwise.  The other 50% feel about as strongly the opposite.  This option has flipped around a lot over the years.

Maybe it could be a user preference?

**Comment by sinkingshriek** [comment]
Hmm. I wonder why people wouldn't want that to be auto focused. Maybe because it doesn't let you use your arrow key to scroll down? Or perhaps in mobile it brings up keyboard? 
I would like to know what the use cases are for the other 50% of people who thinks it's not a good idea.
Regardless, having that in user preference seems kind of overkill but I suppose if others make a really good case for it then I wouldn't be opposed of that idea.

**Comment by aredridel** [comment]
Autofocus destroys accessibility by text to speech, so it cannot be on by default.

**Comment by aredridel** [comment]
And as a user preference, this does in fact seem like overkill, so I'm going to close this issue.

**Comment by aredridel** [comment]
Also: press `/` on the page and the box will be focused.

**Comment by sinkingshriek** [comment]
Fair enough. I knew about the `/` but only after digging into source code I believe. If npm site does end up having lot of shortcuts in future then document like [github](https://help.github.com/articles/using-keyboard-shortcuts/)'s would be tremendous help for new comers. But atm, I'm okay with this.

**Comment by aredridel** [comment]
Indeed! making that discoverable would be good to think about!
